nevergrad/functions/control/core.py

```diff
@@ -40,19 +40,22 @@ class BaseFunction(ExperimentFunction):
     """
 
     def __init__(self, num_rollouts: int, random_state: tp.Optional[int] = None) -> None:
-        super().__init__(self._simulate, p.Array(shape=self.policy_dim))
+        if self.policy_2ndlayer_dim is not None:
```

Comment:
why is that not a parameter ?
Just make a parameter out of it, add it to register_initialization and descritpor, then you do not need a new class. It's the same as for adding noise #844 
also, I guess you don't need to provide a dim, since you don't have a single choice a for it, so make a bool out of it `second_layer: bool`. The simpler the better.

Reply:
We do have choice. Using 17x17 is arbitrary. We could have (N x 376) and (17 x N) for any N.

Reply:
Maybe I can have an optional int. None ==> single_layer. int ==> neural controller.

Reply:
if you do not plan for multiple hidden layers, then that's perfect indeed.